utils_scraping.py

```diff
@@ -181,6 +181,7 @@ def web_files(*urls, dir=os.getcwd(), check=CHECK_NEWER):
         modified = s.head(url).headers.get("last-modified") if check else None
         file = sanitize_filename(url.rsplit("/", 1)[-1])
         file = os.path.join(dir, file)
+        file = ''.join([f for f in file if f not in '?*:<>|']) # Windows Filename Compatibility.
```

Comment:
This is the filename compatibility code for Windows (i.e. just drop chars)... 

I don't know the impact of this while running on other OSes...

Reply:
can you replace the sanitize_filename function with your version?

Reply:
Maybe I'll take a look..

Reply:
- `sanitize_filename(..)` is safe for the filename; and it corrects for unwanted slashes.  
- `not in '?*:<>|'` is safe for the directory path and the filename (yet, it does not correct unwanted slashes).

The error I mentioned (https://github.com/djay/covidthailand/pull/20#issuecomment-855281285) must have come from the directory path, after the full path was combined `os.path.join(dir, file)`.

I would think there is no problem here.

Reply:
The directory  always comes from my code so that is not the problem. I split the last part of the url at the last "/".
If you want to change it santise function name and do the path too its fine but that shouldn't be the problem. Eitherway thats what that function was put there to do so its where that code belongs.

Reply:
Oh I see.... so `s.translate(...)` doesn't work in that way, instead you have to also use `str.maketrans(...)`. i.e.:

```
>>> s = 'json\\query?f=json&where=1%3D1&returnGeometry=false&spatialRel=esriSpatialRelIntersects&outFields=*&resultOffset=0&resultRecordCount=1000&cacheHint=true'
>>> s.translate(str.maketrans({"*": "_", "?": "_", ":": "_", "\\": "_"})) 
'json_query_f=json&where=1%3D1&returnGeometry=false&spatialRel=esriSpatialRelIntersects&outFields=_&resultOffset=0&resultRecordCount=1000&cacheHint=true'
```

Reply:
Corrected and tested with this, which is now in `def sanitize_filename( .. )`
```
filename.translate(str.maketrans({"*": "_", "?": "_", ":": "_", "\\": "_"})) 
```
Runs without errors on local copy on Windows.